File: searching/subarraySums.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

void test() {
    int N, X;
    cin >> N >> X;
    vector<int> T(N);

    for (int i = 0; i < N; ++i) {
        cin >> T[i];
    }

    long long prefixSum = 0;
    long long res = 0;
    map<long long, int> Sum;
    Sum[0] = 1;

    for (int x : T) {
        prefixSum += x;
        res += Sum[prefixSum - X];
        Sum[prefixSum]++;
    }
    
    cout << res << endl;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    test();
    return 0;
}

File: searching/subarraySums.py
import io,os, sys
from itertools import *
import math
import bisect

input = sys.stdin.readline
print = lambda x: sys.stdout.write(str(x))

alphs = "abcdefghijklmnopqrstuvwxyz"
digs = "0123456789"

INT_MIN = -1e10
INT_MAX = 1e10
MODVALUE = 10 ** 9 + 7

def gai():
    return list(map(int, input().strip().split()))

def test():    
    N, X = gai()
    T = gai()
    prefixSum = 0
    res = 0
    prefixMap = {0: 1} 
    for x in T:
        prefixSum += x
        res += prefixMap.get(prefixSum - X, 0)
        prefixMap[prefixSum] = prefixMap.get(prefixSum, 0) + 1
    print(res)

test()